monkey/monkey_island/cc/services/config_schema_service.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, Dict
+
+from common.agent_configuration import AgentConfiguration, PluginConfiguration
+from common.agent_plugins import AgentPlugin, AgentPluginType
+from monkey_island.cc.repositories import IAgentPluginRepository
+
+
+class ConfigSchemaService:
+    """
+    A service for retrieving the agent configuration schema.
+    """
+
+    def __init__(
+        self,
+        agent_plugin_repository: IAgentPluginRepository,
+    ):
+        self._agent_plugin_repository = agent_plugin_repository
+
+    def _create_plugin_schema(self, plugin: AgentPlugin) -> Dict[str, Any]:
+        schema = PluginConfiguration.schema()
+        schema["properties"]["name"]["enum"] = [plugin.plugin_manifest.name]
+        schema["properties"]["options"]["properties"] = plugin.config_schema
+
+        return schema
+
+    def _add_plugins_to_schema(self, schema: Dict[str, Any]):
+        # Note: Can throw RetrievalError, UnknownRecordError
+        plugins = self._agent_plugin_repository.get_plugin_catalog()
+        if plugins:
+            schema["plugins"] = {}
+
+        for (plugin_type, name) in plugins:
+            plugin = self._agent_plugin_repository.get_plugin(plugin_type, name)
+            plugin_schema = self._create_plugin_schema(plugin)
+
+            plugin_type_name = str(plugin_type.name).lower()
+            if plugin_type_name not in schema["plugins"]:
+                schema["plugins"][plugin_type_name] = {"anyOf": []}
```

Comment:
Why do we need "anyOf"? Just `update` the dict in `schema["plugins"][plugin_type_name]` with `{plugin_name: schema}`?

Reply:
The current exploiter classes schema needs `anyOf`